### PR TITLE
[FIX] base_multi_image: Use env in uninstall_hook

### PR DESCRIPTION
**Pull request by lasley**
* Uninstall hook should use new API instead of old, which doesn’t exist anymore

This should fix the Travis fails in OCA/product-attribute#201

### CODE REVIEW

File: base_multi_image/hooks.py [hunk 1]
```diff
@@ -76,9 +76,10 @@ def uninstall_hook_for_submodules(cr, registry, model):
         Model technical name, like "res.partner". All multi-images for that
         model will be deleted
     """
-    Image = registry["base_multi_image.image"]
-    ids = Image.search(cr, SUPERUSER_ID, [("owner_model", "=", model)])
-    Image.unlink(cr, SUPERUSER_ID, ids)
+    env = api.Environment(cr, SUPERUSER_ID, dict())
+    Image = env["base_multi_image.image"]
+    ids = Image.search(env, [("owner_model", "=", model)])
```
Comment (pedrobaeza) [inline base_multi_image/hooks.py]: Better to call the variable `records` or `recs`

**Comment by pedrobaeza** [comment]
`registry` still exists on v10: https://github.com/OCA/OCB/blob/10.0/odoo/modules/registry.py#L23.

What is the value of the argument `registry` passed to the function?

**Comment by lasley** [comment]
@pedrobaeza Registry is valid, I just figured that while the env was there I should use it. Is that not the case?

**Comment by lasley** [comment]
The real error I am circumventing is the `cr, SUPERUSER_ID` being sent into the `.search` method: https://travis-ci.org/OCA/product-attribute/jobs/180247258#L559 - the switch from registry just seemed like bonus while I was there honestly.

**Comment by pedrobaeza** [comment]
Have you tried to use directly `registry`, but without `cr` and `SUPERUSER_ID`?

**Comment by lasley** [comment]
Yeah but an environment is still required as the first argument, so I don't see the different from just using the environment itself. Both of these work:

```
    env = api.Environment(cr, SUPERUSER_ID, dict())
    Image = env["base_multi_image.image"]
    images = Image.search([("owner_model", "=", model)])
    images.unlink()
```

```
    env = api.Environment(cr, SUPERUSER_ID, dict())
    Image = registry["base_multi_image.image"]
    images = Image.search(env, [("owner_model", "=", model)])
    images.unlink()
```

But the former seems cleaner and more OOP IMO

I am about to double check all of this in a new install though just to make sure

**Comment by pedrobaeza** [comment]
OK, thanks for the tests.

**Comment by lasley** [comment]
Hmmm yeah ok so that worked too, without the env. In hindsight I'm actually not sure why it worked the first time around when I passed an env to the `.search`, but it definitely does. 

Simplified version, which I'm pretty sure is what you were thinking of, is pushed. Confirmed it works too 😄

**Review by pedrobaeza — APPROVED** [review]
Yeah, indeed that was my idea.

To the merger: please squash on merge

**Comment by lasley** [comment]
@tedsalmon please review. Solves this - https://travis-ci.org/OCA/product-attribute/builds/190700970 from OCA/product-attribute#201

**Review by StefanRijnhart — APPROVED** [review]

**Pull request merged**